keras_nlp/src/models/clip/clip_text_encoder.py

```diff
@@ -63,64 +60,45 @@ def __init__(
             for _ in range(num_layers)
         ]
         self.layer_norm = layers.LayerNormalization(
-            epsilon=0.00001, dtype=dtype, name="layer_norm"
-        )
-        self.text_projection = layers.Dense(
-            hidden_dim,
-            use_bias=False,
-            dtype=dtype,
-            name="text_projection",
+            epsilon=1e-6, dtype="float32", name="layer_norm"
         )
 
         # === Functional Model ===
-        encoder_token_ids = layers.Input(
-            shape=(sequence_length,), dtype="int32", name="encoder_token_ids"
+        token_id_input = layers.Input(
+            shape=(max_sequence_length,), dtype="int32", name="token_ids"
         )
-        x = self.embedding(encoder_token_ids)
-        encoder_intermediate_output = None
-        # Encoder.
+        x = self.embedding(token_id_input)
+        intermediate_sequence_output = None
         for i, block in enumerate(self.encoder_layers):
             x = block(x)
             if i == intermediate_output_index:
-                encoder_intermediate_output = x
+                intermediate_sequence_output = x
         x = self.layer_norm(x)
-        encoder_output = x
-        if encoder_intermediate_output is not None:
-            encoder_intermediate_output = self.layer_norm(
-                encoder_intermediate_output
-            )
-        # Projection.
-        indices = ops.expand_dims(
-            ops.cast(ops.argmax(encoder_token_ids, axis=-1), "int32"), axis=-1
-        )
-        pooled_output = ops.take_along_axis(x, indices[:, :, None], axis=1)
-        pooled_output = ops.squeeze(pooled_output, axis=1)
-        projection_output = self.text_projection(pooled_output)
+        sequence_output = x
 
-        outputs = {
-            "encoder_sequence_output": encoder_output,
-            "encoder_pooled_output": pooled_output,
-            "encoder_projection_output": projection_output,
-        }
         if intermediate_output_index is not None:
-            outputs["encoder_intermediate_output"] = encoder_intermediate_output
-
+            outputs = {
+                "sequence_output": sequence_output,
+                "intermediate_sequence_output": intermediate_sequence_output,
```

Comment:
What index layer do we normally return here? This is kinda a long name, wonder if we can shorter it somehow.

Reply:
Actually, this should be a special case for downstream models (e.g. StableDiffusion model).
To simplify, I renamed it to `intermediate_output`.

Reply:
sounds good!